runtime/src/accounts_db.rs

```diff
@@ -8269,14 +8249,13 @@ impl AccountsDb {
         ancestors: Option<&Ancestors>,
     ) -> (SnapshotStoragesOne, Vec<Slot>) {
         let mut m = Measure::start("get slots");
-        let slots_and_storages = self
+        let mut slots_and_storages = self
             .storage
             .iter()
-            .filter_map(|entry| {
-                let slot = *entry.key() as Slot;
+            .filter_map(|(slot, store)| {
                 requested_slots
                     .contains(&slot)
-                    .then_some((slot, Arc::clone(entry.value())))
```

Comment:
Here was a clone per store we were doing in the only call chain that matters, when we are getting snapshot storages.
It is true that when generating an incremental snapshot, we are now cloning every store in the map even though we only need a fraction of those stores. We could pass in a filter by slot closure to `iter()` so that we only clone the stores that match a slot criteria.